torch/optim/radam.py

```diff
@@ -405,29 +433,61 @@ def _multi_tensor_radam(
         # Delete the local intermediate since it won't be used anymore to save on peak memory
         del grouped_grads
 
-        rect = [
-            _dispatch_sqrt(
-                (rho_t - 4)
-                * (rho_t - 2)
-                * rho_inf
-                / ((rho_inf - 4) * (rho_inf - 2) * rho_t)
-            )
-            if rho_t > 5
-            else 0
-            for rho_t in rho_t_list
-        ]
-        unrectified = [0 if rect > 0 else 1.0 for rect in rect]
-
-        bias_correction1 = [1 - beta1 ** _get_value(step) for step in grouped_state_steps]
-        unrect_step_size = _stack_if_compiling([(lr * rect / bc) * -1 for rect, bc in zip(unrectified, bias_correction1)])
-        bias_correction2_sqrt_times_rect_step_size = [
-            _dispatch_sqrt(1 - beta2 ** _get_value(step)) * (lr * rect / bc) * -1
-            for step, rect, bc in zip(grouped_state_steps, rect, bias_correction1)
-        ]
+        if capturable:
+            num = torch._foreach_sub(rho_t_list, 4)
+            sub2 = torch._foreach_sub(rho_t_list, 2)
+            torch._foreach_mul_(num, sub2)
+            torch._foreach_mul_(num, rho_inf)
+            rho_inf = ((rho_inf - 4) * (rho_inf - 2))
+            denom = torch._foreach_mul(rho_t_list, rho_inf)
+            torch._foreach_div_(num, denom)
+            torch._foreach_sqrt_(num)
+
+            # TODO(mlazos): we should try and get a foreach_where op https://github.com/pytorch/pytorch/issues/117884
+            rect = [torch.where(rho_t > 5, n, torch.zeros_like(n)) for n, rho_t in zip(num, rho_t_list)]
+            unrect_step_size = [torch.where(rect > 0, torch.zeros_like(rect), torch.ones_like(rect)) for rect in rect]
```

Comment:
btw, torch.where now supports python scalars, so probably `[torch.where(rect > 0, 0.0, 1.0) for rect in rect]` should work or even `(rect > 0).type_as(rect)`

Reply:
I actually need a full tensor of the value for the usage below actually

Reply:
The output of torch.where is still a full tensor, even if you pass in scalars instead of zeros_like/ones_like, right?

Currently zeros_like and ones_like are materialized just for passing into torch.where

Reply:
Ah you're right, done

Reply:
and maybe even `(rect <= 0).type_as(rect)` would be correct here... I don't know if any of these is better for torch.compile...